callback.php

```diff
@@ -108,6 +103,37 @@
     echo 'Secret not matching';
 }
 
+function insertTXIDIntoPaymentDetails($order, $txid, $bits_payed)
+{
+    //Get amount payed in fiat currency
+    $currency = new Currency((int) $order->id_currency);
+    $blockonomics = new Blockonomics();
+    $url = Configuration::get('BLOCKONOMICS_PRICE_URL') . $currency->iso_code;
+    $price = $blockonomics->doCurlCall($url)->data->price;
+    $total = $price * $bits_payed * 1.0e-8;
+    $rounded_total = round($total, 2);
+
```

Comment:
There are two problems with this:

1. We should not do a curl in the first place as confirmation will take time and by that time price may have fallen 20%. The price is effectively locked when the order is created, we should use the same price
2. If user has the exact BTC amount is a paid, the exact amount of fiat currency should be credited (rounding may create problems)

Reply:
1. The reason the curl call is being made is to calculate the value that will be displayed in the amount column of the payment details. We could change the amount to reflect the btc that is being transferred in the transaction. Otherwise, we need to store the btc/currency somewhere. 
![image](https://user-images.githubusercontent.com/36883813/116364867-9e77f600-a83f-11eb-91ca-a700b6a7a436.png)

2. I've removed the rounding.

Reply:
Nevermind, I figured out that it's possible to use the data we have in blockonomics_orders to calculate the value in the local currency.
![image](https://user-images.githubusercontent.com/36883813/116366049-d6cc0400-a840-11eb-9ae2-6cbafe1dc523.png)